### ISSUE TITLE
Gallery images don't load in Magnific Popup modal

### ISSUE THREAD

**Issue by axis80** [madoublet/respond#407]
In previous versions of Respond, clicking on a gallery image loaded it in a nice Magnific Popup modal.  With the current dev branch, the user is simply linked to the raw image file URL.

**Comment by SylvainMoingeon** [comment]
Hello, 

I have sometimes this bug then it resolves by itself (I don't do anything !) and then the bug appear again. Sometimes on Chrome, sometimes on firefox. this is really weird.
https://groups.google.com/forum/#!topic/respond-cms/2kGEYK5iXrc

**Comment by axis80** [comment]
Sylvain, I think this is a clue that will solve the mystery.  I believe it is a race condition.  The gallery images are rendered into the DOM dynamically.  What happens is that sometimes they haven't been rendered yet, at the time the magnific popup plugin is initialized.  So, sometimes it works and other times it doesn't, depending on which script finishes first.

I will see if I can fix this later.

**Comment by Maximumz** [comment]
I have noticed this on 5.3, the tabs appear to be affected also.

**Comment by axis80** [comment]
I don't think it matters whether the scripts are loaded in the head vs the bottom of the page.  The problem is that the web components take time to render and become part of the DOM.  That takes an indeterminate amount of time, and partly depends on the number of components you've dragged onto the page.  The initialization of Magnific Popup happens separately, and there is no guarantee it won't execute before the web components finish rendering.

I have run into this a lot in the past while writing JS for custom themes.  I should be able to fix it by adding a setTimeout wrapper around the Magnific Popup init, so it will wait until the gallery image DOM nodes appear.

**Comment by axis80** [comment]
Hmm, I just started to add the setTimeout code, and I realized that it is not an optimal solution.  It will wait until a respond-gallery appears in the DOM, but it has no way of knowing how many respond-galleries there are in total.  So it's possible it could fire after the first one was inserted into the DOM, but miss any additional galleries on the page.

My thinking now is that it's best to change it to watch for new DOM nodes to be inserted and attach magnific popup to each one as it appears.  Looks like this library would do the trick nicely:  

https://github.com/uzairfarooq/arrive/

I have already spent more time on this than I have available.  I am leaving this conversation in place to document the problem and proposed solution.  If someone else wants to take this information and create a pull request I'll accept it quickly.  Otherwise, it may be a couple weeks until I can work on this again.
